### DISCUSSION TITLE
set permissions/ roles between relationships

### DISCUSSION THREAD

**Discussion by jcc5018** [spatie/laravel-permission#2241] [Q&A]
If i want to be able to define roles between two models such as a user to user relationship, do you have an idea how I may be able to use or alter your package to accommodate this? 

I currently have what I call friend groups which would relate to roles, in which groups and or individual friends can have separate permissions set such as viewing portfolio or allow chat. 

So my first thought would be a middleware that checks the active user to the user being accessed to see if they are allowed to perform an action. But im not sure how to define the 'user_see_profile'  and similar permissions on a m:m relationship opposed to the entire model. 

I'm still a fairly new developer, so any guidance as  to how to tackle this would be appreciated. 

As of now this is basically to control social interactions between users for use in blocking, or restricting what various people can see. So perhaps their is another package that would make this easier?

**Comment by angeljqv** [comment]
Maybe another package is what you are looking for
